src/aggregator/client/conn.go

```diff
@@ -192,6 +226,15 @@ func (c *connection) connectWithLock() error {
 		}
 	}
 
+	if c.tls.TLSEnabled() {
+		conn, err = c.upgradeToTLS(conn)
+		if err != nil {
```

Comment:
should here be a check for permissive mode config value and if it's set to `true` we fallback to insecure connection and log the error?

Reply:
It is relevant for a server only. After establishing a connection, clients should know they should send data or initiate a TLS handshake. I don't expect such an error because the client should be switched to TLS after the server.